module-2/lab-probability-distribution/your_code/main.ipynb

```diff
@@ -12,7 +12,7 @@
   },
```

Comment:
- Sometimes it's useful to seed the random number generator. Seeding a pseudo-random number generator gives it its first "previous" value. Each seed value will correspond to a sequence of generated values for a given random number generator. That is, if you provide the same seed twice, you get the same sequence of numbers twice. https://machinelearningmastery.com/how-to-generate-random-numbers-in-python/
- Regarding this: "We have received True for all samples, that means that we can't reject the null 
hypothesis in any of the cases. We can assume our samples are normally distributed". In statistics, we say we can't reject the null hypothesis. The fact that we can't reject doesn't mean we accept normality. This could sound counterintuitive but it's the way it works in statistics. In spanish, Esto lo decimos porque a pesar de un resultado positivo del estadístico de contraste podría no ser normal. Esto lo ves claramente en el primer histograma tamaño de la muestra 5, el p-value te dice que no puedes rechazar la hipótesis nula de normalidad y, sin embargo, el histograma muestra algo que no tiene nada que ver con una normal. Aquí podríamos graficar un gráfico de quantiles y, probablemente, veríamos, al igual que en el histograma, que los datos no vienen de una distribución normal.
- Normality tests are quite sensitive to sample size thus for a sample size below a couple of thousands it should be a better idea to evaluate normality through a plot, in particular a quantile-quantile plot (QQ plot). A Q-Q plot is a scatterplot created by plotting two sets of quantiles against one another. It plots the quantiles of our sample vs the theoretical quantiles of the distribution type we want to test (this case a normal distribution) How to do this in Python? https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.probplot.html.
- In case you are interested, there are 2 other normality tests (additional to D'Agostino k^2 test, normaltest in Python) that are very used in to test normality of our data. These are Kolmogorov-Smirnof and Shapiro-Wilk tests. I encourage you to deepen on these type of tests. In case you are interested, I can provide you some useful resources.
- Interpretation of the CLT: take a look on these 2 articles (the most clear explanations I've found). You can replicate in Python some of the examples they include.
https://towardsdatascience.com/understanding-the-central-limit-theorem-642473c63ad8.
https://www.statisticshowto.datasciencecentral.com/probability-and-statistics/normal-distributions/central-limit-theorem-definition-examples/
- Regarding this, good try but the main conclusion has not been highlighted. As larger the sample , the binomial distribution looks more like a normal distribution. Why?
Normal distribution can be used as an approximation of Binomial and Poisson distributions under certain circumstances. Generally speaking:
Binomial: X ~ B(n, p) and if n is large and/or p is close to ½, then X is approximately N(np, npq)
Poisson: X ~ Po(l) then for large values of l, X ~ N(l, l) approximately.
However, we must be careful because binomial and poisson distributions are discrete while Normal distribution is continuous. We must use a continuity correction.
```Python
'''
It is observed that since we are dealing with groups composed of many individuals, all plots show a PMF line
that starts at 0 (or at least, fairly close to 0), which makes sense considering that it will be quite 
difficult to have nobody clicking on the Ad and also end at probability 0ish.
In the three plot we observe a sudden very intense increase on the probability followed by a sudden very 
intense decrease that tends to occupy a smaller proportion in the range of clicks the larger the sample is.
This peak in the probability happens around the 1/10 of the range of click, progessively moving to the left 
whit larger samples.
'''
- As simple as this: 
```Python 
fig, ax = plt.subplots(1, 1)
mu=300
x = np.arange(0,1000)
ax.plot(x, sci.poisson.pmf(x, mu), 'bo', ms=8, label='poisson pmf')
```
You can observe the same as in binomial challenge, as larger the sample in a poisson distribution more similar to a normal distribution.